### ISSUE TITLE
Clusterresourcebindings informer watching fails with goroutine hanging after a while

### ISSUE THREAD

**Issue by yanfeng1992** [karmada-io/karmada#4143]
**What happened**:

 After a few days of normally running, we can't receive any events when the resource clusterresourcebindings changes(other resources and apiserver are normal). We found out that the corresponding goroutine has been hung for a long time:

The same problem also occurs on the Work resource

![image](https://github.com/karmada-io/karmada/assets/89568107/00e818df-5d83-4416-a331-c74b5c5d7287)

**What you expected to happen**:

We can receive events when the resource clusterresourcebindings changes.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Karmada version: 1.5
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

**Comment by yanfeng1992** [comment]
Has anyone in the community encountered the same problem? @RainbowMango

**Comment by RainbowMango** [comment]
No, I hadn't heard of this before.
I can not figure out which piece of code is hanging from the picture.
Do you have any clue about it? Sorry for the late response, I missed it.

**Comment by yanfeng1992** [comment]
>  I can not figure out which piece of code is hanging from the picture. Do you have any clue about it?

https://github.com/karmada-io/karmada/blob/98e655fc552b2987c3f2d2a061007889ce8be536/pkg/generated/clientset/versioned/typed/work/v1alpha2/clusterresourcebinding.go#L78

**Comment by RainbowMango** [comment]
Do you know where the goroutine is from where? From the controller-runtime?

**Comment by yanfeng1992** [comment]
https://github.com/karmada-io/karmada/blob/98e655fc552b2987c3f2d2a061007889ce8be536/pkg/controllers/binding/cluster_resource_binding_controller.go#L148

I think the source is here. After karmada has been running for a period of time, the cluster_resource_binding_controller in karmada-controller-manager does not watch any changes to crb. I don't know whether it is a problem with controller-runtime or karmada.

**Comment by RainbowMango** [comment]
Yeah, I can feel that this issue is hard to reproduce and hard to find the root cause from code review.

Do you have the pprof file? 

> we can't receive any events when the resource clusterresourcebindings changes(

How do you distinguish whether it didn't receive the event or didn't reconcile?
Were there many resources in the system at that time? I wonder if it could just be a slow reconcile?

**Comment by yanfeng1992** [comment]
@whitewindmills Can we provide any more information?

**Comment by whitewindmills** [comment]
it's not a slow reconcile but a permanent hang. from this piture, the informer goroutine has been hung for 2212 minutes. from the logs we cannot see any resource being processed, and the only way to solve the problem is to restart the relevant pods.

**Comment by whitewindmills** [comment]
Confusingly, this phenomenon only appears on `rb/crb`, but never on other resources, so we doubt whether it is related to Karmada.

**Comment by RainbowMango** [comment]
Have you collected metrics of `karmada-controller-manager`? And logs?

By the way.
I just quickly went through the release notes of `controller-runtime`, not find any related bug fix. I don't have a clue yet.
